GherkinSyncTool.Synchronizers.AzureDevOps/AzureDevopsSynchronizer.cs

```diff
@@ -38,64 +40,148 @@ public void Sync(List<IFeatureFile> featureFiles)
+
+            var patchDocumentId = -1;
 
             foreach (var featureFile in featureFiles)
             {
                 foreach (var scenario in featureFile.Document.Feature.Children.OfType<Scenario>())
                 {
-                    var tagId = scenario.Tags.FirstOrDefault(tag => tag.Name.Contains(_gherkinSyncToolConfig.TagIdPrefix));
-                    
+                    var tagIds = scenario.Tags.Where(tag => tag.Name.Contains(_gherkinSyncToolConfig.TagIdPrefix)).ToList();
+
+                    if (tagIds.Count > 1)
+                    {
+                        Log.Warn($"There are multiple tag IDs in the scenario. Using the last one ID. Scenario name: {scenario.Name}");
+                    }
+
+                    var tagId = tagIds.LastOrDefault();
+
                     // Create test case for feature file that first time sync, no tag id present.  
                     if (tagId is null)
                     {
-                        var testCasePatchDocument = _caseContentBuilder.BuildTestCaseDocument(scenario, featureFile, caseToCreateId--);
-                        var testCaseBatchRequest = _azureDevopsClient.CreateTestCaseBatchRequest(testCasePatchDocument);
-                        caseToCreate.Add(testCaseBatchRequest);
+                        var testCasePatchDocument = _caseContentBuilder.BuildTestCaseDocument(scenario, featureFile, patchDocumentId--);
+                        var testCaseBatchRequest = _azureDevopsClient.BuildCreateTestCaseBatchRequest(testCasePatchDocument);
+                        witBatchRequests.Add(testCaseBatchRequest);
+                    }
+
+                    // Update scenarios that have tag id
+                    if (tagId is not null)
+                    {
+                        var caseId = (int)GherkinHelper.GetTagIdUlong(tagId);
+                        if (fullTestCasesIdList.Contains(caseId))
+                        {
+                            var testCasePatchDocument = _caseContentBuilder.BuildTestCaseDocument(scenario, featureFile);
+                            var testCaseBatchRequest = _azureDevopsClient.BuildUpdateTestCaseBatchRequest(caseId, testCasePatchDocument);
+
+                            testCasesToUpdateFromTheFeatureFiles.Add(caseId, testCaseBatchRequest);
+                        }
+                        else
+                        {
+                            Log.Warn($"Test case with id {caseId} not found. Missing case will be recreated");
+                            var testCasePatchDocument = _caseContentBuilder.BuildTestCaseDocument(scenario, featureFile, patchDocumentId--);
+                            var testCaseBatchRequest = _azureDevopsClient.BuildCreateTestCaseBatchRequest(testCasePatchDocument);
+                            
+                            TextFilesEditMethods.ReplaceTextInTheFileRegex(featureFile.AbsolutePath,$@"\s*{GherkinHelper.FormatTagId(caseId.ToString()).Trim()}\s*","");
```

Comment:
looks a bit convoluted, let's consider extracting second parameter into the variable

Reply:
Done